dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart

```diff
@@ -121,6 +123,113 @@ void _fallbackToManual(String error) {
   }
 }
 
+/// Visualizes results on the Web page for manual inspection.
+void _printResultsToScreen(Profile profile) {
+  html.document.body.remove();
+  html.document.body = html.BodyElement();
+  html.document.body.appendHtml('<h2>${profile.name}</h2>');
+
+  profile.scoreData.forEach((String scoreKey, Timeseries timeseries) {
+    html.document.body.appendHtml('<h2>$scoreKey</h2>');
+    html.document.body.appendHtml('<pre>${timeseries.computeStats()}</pre>');
+    html.document.body.append(TimeseriesVisualization(timeseries).render());
+  });
+}
+
+/// Draws timeseries data and statistics on a canvas.
+class TimeseriesVisualization {
+  TimeseriesVisualization(this._timeseries) {
+    _stats = _timeseries.computeStats();
+    _canvas = html.CanvasElement();
+    _screenWidth = html.window.screen.width;
+    _canvas.width = _screenWidth;
+    _canvas.height = (_kCanvasHeight * html.window.devicePixelRatio).round();
+    _canvas.style
+      ..width = '100%'
+      ..height = '${_kCanvasHeight}px'
+      ..outline = '1px solid green';
+    _ctx = _canvas.context2D;
+
+    // The amount of vertical space available on the chart. Because some
+    // outliers can be huge they can dwarf all the useful values. So we
+    // limit it to 1.5 x the biggest non-outlier.
+    _maxValueChartRange = 1.5 * _stats.samples
+      .where((AnnotatedSample sample) => !sample.isOutlier)
+      .map<double>((AnnotatedSample sample) => sample.magnitude)
+      .fold<double>(0, math.max);
+  }
+
+  static const double _kCanvasHeight = 200;
+
+  final Timeseries _timeseries;
+  TimeseriesStats _stats;
+  html.CanvasElement _canvas;
+  html.CanvasRenderingContext2D _ctx;
+  int _screenWidth;
+
+  // Used to normalize benchmark values to chart height.
+  double _maxValueChartRange;
+
+  /// Converts a sample value to vertical canvas coordinates.
+  ///
+  /// This does not work for horizontal coordinates.
+  double _normalized(double value) {
+    return _kCanvasHeight * value / _maxValueChartRange;
+  }
+
+  /// A utility for drawing lines.
+  void drawLine(num x1, num y1, num x2, num y2) {
+    _ctx.beginPath();
+    _ctx.moveTo(x1, y1);
+    _ctx.lineTo(x2, y2);
+    _ctx.stroke();
+  }
+
+  /// Renders the timeseries into a `<canvas>` and returns the canvas element.
+  html.CanvasElement render() {
+    _ctx.translate(0, _kCanvasHeight * html.window.devicePixelRatio);
+    _ctx.scale(1, -html.window.devicePixelRatio);
+
+    final double barWidth = _screenWidth / _stats.samples.length;
+    double xOffset = 0;
+    for (int i = 0; i < _stats.samples.length; i++) {
+      final AnnotatedSample sample = _stats.samples[i];
+
+      if (sample.isWarmUpValue) {
+        _ctx.fillStyle = 'rgba(200,200,200,1)';
+        _ctx.fillRect(xOffset, 0, barWidth, _normalized(_maxValueChartRange));
+      }
+
+      if (sample.magnitude > _maxValueChartRange) {
+        _ctx.fillStyle = 'rgba(100,50,100,0.8)';
```

Comment:
Is editor showing colors? if not placing comment/name of color/meaning would help.

Reply:
Done.